google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/AppEngineFlexibleSupportProvider.java

```diff
@@ -165,6 +128,50 @@ public static void addSupport(@NotNull AppEngineFlexibleFacet facet,
     setupDeploymentRunConfiguration(facet.getModule());
   }
 
+  /**
+   * Generates an app.yaml configuration file in the src/main/appengine directory. If an app.yaml
+   * already exists it will not overwrite the file.
+   */
+  private static void generateAppYaml(@NotNull AppEngineFlexibleFacet facet,
+      @NotNull VirtualFile contentRoot) {
+    Project project = facet.getModule().getProject();
+    FileTemplate appYamlTemplate = FileTemplateManager.getInstance(project)
+        .getInternalTemplate(AppEngineTemplateGroupDescriptorFactory.APP_YAML_TEMPLATE);
+
+    VirtualFile virtualFile = contentRoot.findFileByRelativePath("src/main");
+
+    if (virtualFile != null) {
+      PsiDirectory directory = PsiManager.getInstance(project)
+          .findDirectory(virtualFile);
+
+      if (directory != null) {
+        PsiDirectory appEngineDirectory;
+        try {
+          directory.checkCreateSubdirectory(CONFIG_DIR_NAME);
+          appEngineDirectory = directory.createSubdirectory(CONFIG_DIR_NAME);
+        } catch (IncorrectOperationException ioe) {
+          // checkCreateSubdirectory threw an exception suggesting that the directory may already
+          // exist. Skip creating the directory and attempt to write file.
+          appEngineDirectory = directory.findSubdirectory(CONFIG_DIR_NAME);
+        }
+
+        if (appEngineDirectory != null
+            && FileTemplateUtil.canCreateFromTemplate(
+            new PsiDirectory[]{appEngineDirectory}, appYamlTemplate)) {
+          try {
+            FileTemplateUtil.createFromTemplate(appYamlTemplate, "app.yaml",
+                FileTemplateManager.getInstance(project).getDefaultProperties(),
+                appEngineDirectory);
+          } catch (Exception e) {
+            // If the file already exists, this exception will be thrown by createFromTemplate
+            logger.warn("Failed to create app yaml from template. " + e.getMessage());
```

Comment:
I could add an explicit check to if the file exists prior to calling createFromTemplate.

But we then still have to problem of having to handle the exception thrown by createFromTemplate (and it does throw Exception, not a subclass) which could happen for various reasons.